security/src/main/java/io/micronaut/security/audit/UserAuditingEntityEventListener.java

```diff
@@ -83,23 +82,30 @@ public boolean preUpdate(@NonNull EntityEventContext<Object> context) {
         };
     }
 
-    private void autoPopulateUserIdentity(@NonNull EntityEventContext<Object> context, boolean isUpdate) {
+    private void populate(@NonNull EntityEventContext<Object> context,
+                          @NonNull Class<? extends Annotation> listenerAnnotation) {
         if (securityService.isAuthenticated()) {
-            final RuntimePersistentProperty<Object>[] applicableProperties = getApplicableProperties(context.getPersistentEntity());
-            for (RuntimePersistentProperty<Object> persistentProperty : applicableProperties) {
-                if (isUpdate) {
-                    if (!persistentProperty.getAnnotationMetadata().booleanValue(AutoPopulated.class, AutoPopulated.UPDATEABLE).orElse(true)) {
-                        continue;
-                    }
-                }
-
-                final BeanProperty<Object, Object> beanProperty = persistentProperty.getProperty();
-                getCurrentUserIdentityForProperty(beanProperty).ifPresent(identity -> context.setProperty(beanProperty, identity));
-            }
+            securityService.getAuthentication().ifPresent(authentication -> {
```

Comment:
I think at this point the outermost `if (securityService.isAutheticated())` is redundant and can be removed.

Reply:
[I have removed the outermost `if`](https://github.com/micronaut-projects/micronaut-security/pull/1539/commits/16f3cd7a9c793e6b88ecf3aae59db2eeac8f170b)